ocs_ci/utility/utils.py

```diff
@@ -406,31 +428,32 @@ def run_cmd(cmd, secrets=None, timeout=600, ignore_error=False, **kwargs):
     log.info(f"Executing command: {masked_cmd}")
     if isinstance(cmd, str):
         cmd = shlex.split(cmd)
-    r = subprocess.run(
+    completed_process = subprocess.run(
         cmd,
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE,
         stdin=subprocess.PIPE,
         timeout=timeout,
         **kwargs
     )
-    masked_stdout = mask_secrets(r.stdout.decode(), secrets)
-    if len(r.stdout) > 0:
+    masked_stdout = mask_secrets(completed_process.stdout.decode(), secrets)
+    if len(completed_process.stdout) > 0:
         log.debug(f"Command stdout: {masked_stdout}")
     else:
         log.debug("Command stdout is empty")
-    masked_stderr = mask_secrets(r.stderr.decode(), secrets)
-    if len(r.stderr) > 0:
+
+    masked_stderr = mask_secrets(completed_process.stderr.decode(), secrets)
+    if len(completed_process.stderr) > 0:
         log.warning(f"Command stderr: {masked_stderr}")
     else:
         log.debug("Command stderr is empty")
-    log.debug(f"Command return code: {r.returncode}")
-    if r.returncode and not ignore_error:
+    log.debug(f"Command return code: {completed_process.returncode}")
+    if completed_process.returncode and not ignore_error:
         raise CommandFailed(
             f"Error during execution of command: {masked_cmd}."
             f"\nError is {masked_stderr}"
         )
-    return masked_stdout
+    return completed_process
```

Comment:
please fix the returned value in the doc

Reply:
Already done 👍

Reply:
Also, as we need to mask the output, I wonder if it will be better to return a tuple of (rc, out, err) all masked

Reply:
I think it can actually be valuable for us to have programmatic access to the unmasked variables. It's still masked in the logs, and thus, the only way to "leak" any values would be to intentionally log again what we already logged in `exec_cmd`. My mindset is to keep the object as untouched as possible, because every processing we choose to do causes a loss of information, which we might potentially need down the road (which is the reason for this PR).

Reply:
@MeridianExplorer how about adding an example for how to unpack completed_process?

Reply:
@ebenahar Added :+1: